lib/model/CTokenListDataCategorizerBase.cc

```diff
@@ -200,15 +200,18 @@ bool CTokenListDataCategorizerBase::createReverseSearch(int categoryId,
             return false;
         }
 
+        // Ensure all reference arguments are set before returning true
+        wasCached = false;
+        maxMatchingLength = 0;
```

Comment:
`wasCached` and `maxMatchingLength` are unset in the earlier return path `if (m_ReverseSearchCreator == nullptr)`.

I guess they're unused if the return value is false, but it strikes me as safer to just initialise them like this at the very start of the function.

Reply:
Yes, the idea is that there are no guarantees on the reference variables if the return value is false.  Still, I'll move them to the beginning so future changes don't accidentally introduce a problem.